src/main/haskell/kore/test/Test/Kore/Step/Function/Registry.hs

```diff
@@ -0,0 +1,274 @@
+import           Kore.IndexedModule.IndexedModule
+                 ( KoreIndexedModule )
+import           Kore.IndexedModule.MetadataTools
+                 ( MetadataTools (..), extractMetadataTools )
+import           Kore.Predicate.Predicate
+                 ( makeTruePredicate )
+import           Kore.Step.AxiomPatterns
+                 ( koreIndexedModuleToAxiomPatterns )
+import           Kore.Step.ExpandedPattern
+                 ( CommonExpandedPattern, ExpandedPattern (..) )
+import qualified Kore.Step.ExpandedPattern as ExpandedPattern
+import           Kore.Step.Function.Data
+                 ( CommonApplicationFunctionEvaluator )
+import           Kore.Step.Function.Registry
+import qualified Kore.Step.OrOfExpandedPattern as OrOfExpandedPattern
+import           Kore.Step.Simplification.Data
+                 ( evalSimplifier )
+import qualified Kore.Step.Simplification.ExpandedPattern as ExpandedPattern
+import           Kore.Step.StepperAttributes
+
+import Test.Kore.ASTVerifier.DefinitionVerifier
+import Test.Kore.Comparators ()
+
+updateAttributes :: Attributes -> KoreSentence -> KoreSentence
+updateAttributes attrs = applyUnifiedSentence updateAttrs updateAttrs
+  where
+    updateAttrs
+        :: MetaOrObject level
+        => Sentence level UnifiedSortVariable UnifiedPattern Variable
+        -> KoreSentence
+    updateAttrs (SentenceSymbolSentence ss) =
+        constructUnifiedSentence SentenceSymbolSentence
+            (ss { sentenceSymbolAttributes = attrs })
+    updateAttrs _ = error "unsupported non-symbol sentence"
+
+sortVar :: SortVariable Object
+sortVar = SortVariable (testId "R")
+
+sortVarS :: Sort Object
+sortVarS = SortVariableSort sortVar
+
+sortS :: Sort level
+sortS = SortActualSort (SortActual (testId "S") [])
+
+fHead, gHead, sHead, tHead :: SymbolOrAlias level
+fHead = groundHead "f" AstLocationTest
+gHead = groundHead "g" AstLocationTest
+sHead = groundHead "s" AstLocationTest
+tHead = groundHead "t" AstLocationTest
+
+testDef :: KoreDefinition
+testDef = simpleDefinitionFromSentences
+            (ModuleName "test")
+            [ simpleSortSentence (SortName "S")
+            , simpleObjectSymbolSentence (SymbolName "s") (SortName "S")
+            , simpleObjectSymbolSentence (SymbolName "t") (SortName "S")
+            , updateAttributes
+                (Attributes [functionAttribute, constructorAttribute])
+                (simpleObjectSymbolSentence (SymbolName "f") (SortName "S"))
+            , updateAttributes
+                (Attributes [functionAttribute, constructorAttribute])
+                (simpleObjectSymbolSentence (SymbolName "g") (SortName "S"))
+            , asSentence
+                (SentenceAxiom
+                    { sentenceAxiomParameters = [asUnified sortVar]
+                    , sentenceAxiomAttributes = Attributes []
+                    , sentenceAxiomPattern =
+                        (patternPureToKore
+                            ( Implies_ sortVarS
```

Comment:
You can leave it as-is, but, FWIW, the mk* constructors will automatically infer sorts and IIRC you can force the sort at the top to be sortVarS with some function I don't remember.

Reply:
Well, I thought about it, but for `mk*` constructors I need to provide some  `SortTools` which might be more cumbersome, as they would be for the definition I'm trying to build.
Of course, I could create another definition just with the syntax, or mock them, but it seems more work than just spelling the sorts.